src/hotspot/share/runtime/synchronizer.inline.hpp

```diff
@@ -45,11 +45,7 @@ inline ObjectMonitor* ObjectSynchronizer::read_monitor(Thread* current, oop obj,
 inline void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current) {
   assert(current == Thread::current(), "must be");
 
-  if (LockingMode == LM_LIGHTWEIGHT) {
     LightweightSynchronizer::enter(obj, lock, current);
```

Comment:
In the further RFE, we should remove these dispatch functions too.

Reply:
So to be clear, we should probably have 2+ RFEs that follow this.  One to remove the cpu specific names like displaced header, one to remove the loom interactions and held_monitor_count, and another to remove the dispatches to LightweightSynchronizer and make ObjectSynchronizer's role in object locking clear.